src/ioc_scan/stix_extract.py

```diff
@@ -0,0 +1,142 @@
+from docopt import docopt
+from stix.core import STIXPackage
+
+
+def extract_stix_info(stix_file):
+    """
+    Extract valuable information (IP addresses, hashes, FQDNs, and URLs) from a STIX file.
+
+    Prioritize hashes based on their type: SHA256 > SHA1 > MD5.
+
+    Args:
+        stix_file (str): path to the STIX file to parse
+
+    Returns:
+        tuple: a tuple containing four lists - one for IP addresses, one for hashes, one for FQDNs, and one for URLs.
+    """
+    # Load the STIX package from the XML file
+    try:
+        stix_package = STIXPackage.from_xml(stix_file)
+    except Exception as e:
+        print(f"Error parsing STIX file: {e}")
+        raise e
+
+    # Initialize lists to store IP addresses, hashes, FQDNs, and URLs
+    ip_addresses = []
+    hashes = []
+    fqdns = []
+    urls = []
+
+    # Define hash type priority. Lower value means higher priority.
+    hash_priority = OrderedDict([("SHA256", 0), ("SHA1", 1), ("MD5", 2)])
+
+    # Iterate over each indicator in the STIX package
+    for indicator in stix_package.indicators:
+        for observable in indicator.observables:
+            object_type = observable.object_.properties._XSI_TYPE
+            if object_type == "AddressObjectType":
+                # Convert cybox.common.properties.String to str
+                ip_addresses.append(str(observable.object_.properties.address_value))
+            elif object_type == "FileObjectType":
+                hashes_dict = observable.object_.properties.hashes
+                if hashes_dict:
+                    best_hash = None
+                    best_priority = float("inf")
+                    for h in hashes_dict:
+                        if (
+                            h.type_.value in hash_priority
+                            and hash_priority[h.type_.value] < best_priority
+                        ):
+                            best_hash = str(h.simple_hash_value.value)  # Convert to str
+                            best_priority = hash_priority[h.type_.value]
+                    if best_hash is not None:
+                        hashes.append(best_hash)
+            elif object_type == "DomainNameObjectType":
+                # Convert cybox.common.properties.String to str
+                fqdns.append(str(observable.object_.properties.value.value))
+            elif object_type == "URIObjectType":
+                # Convert cybox.common.properties.String to str
+                urls.append(str(observable.object_.properties.value.value))
+
+    return ip_addresses, hashes, fqdns, urls
+
+
+def sort_ip_address(ip):
+    """
+    Take an IP address as input and return a tuple that can be used for sorting.
+
+    Args:
+        ip (str): an IP address
+
+    Returns:
+        tuple: a tuple containing two elements - the IP version (int) and the integer representation of the IP address (int).
+    """
+    ip_version = ipaddress.ip_address(ip).version
+    ip_int = int(ipaddress.ip_address(ip))
```

Comment:
No need to create `ipaddress.ip_address(ip)` twice here.

Reply:
See: [a040345](https://github.com/cisagov/ioc-scanner/pull/48/commits/a040345bf99cf25d2e6b5f4bc54fb55a5f386291)